### PR TITLE
Mouse surgery form

### PR DESCRIPTION
**Pull request by david-roper**
Version 1.0 of the mouse surgery form

closes issue #21 

form can be seen [here](https://playground.opendatacapture.org/?files=NobwRAxg9gdgLgU3mAXGA9AKgAQIM4A2AlvALQAmReAhgEYELYAOCATgGYIRxGxVzo8UVnFJRaAKy5w82TOgA6MJdBh442ENnIJ2JBAEk1cVgFcAtkg0BfbAF5s1AO7UiGouabC4ACgDk6GbwHgjoAG4AjOgAAlAsMOTUcNQQ1ExwpqyhQTyWpNBZ6CQ6AB4AdBJ4fgCUANwqsOqa2ABe2LYOzq7unt7%2BgabBluFRLVDk0QDMZQBM0yVFCQjllTX1yjDsg9y8MNgQWUkIACII8TrwAGJECATkADwAKgB8Pnq35CjYjwA02OwwL4%2BEwIJKWeCPACeLC%2B6lYJAA5tV7M9sLQoFAGNQYMiQEpsNgshlWHs8XsCdgANbFL5%2BciQmDUcxECB%2BRyyVTqH74ik6Jh4L7APwgsFWKEsPwAXXZ%2B0acG55IJWSWrCBiWSXy0Irg4Lg4oQsJMiPayLsqLJFIpRHY2DeMB86uoZW1uv11VxPMtSoQxL27zu6y97U9FKJmT2MFMBAIgYp1h51nW8Y2yy8Im0umoUY0Oj0MEMxjMup8FqpNOwfkuAHkAEoAWT8CoJBGxCNM1ARBorSEbPOSCIF2CFAGVMp3WJDGxWAGo%2B7AAYWoWSnfgA6lBBuRvocdVYV1WwkubtwoOZJ5Km9gSIgSdQCJqQwhKDxYF8IpeCYzLLS61WAKrDgAogA%2BsOf41gA4oBNYAJrAdW9Z%2BAml6qIg8APoq2AumK0JdqWlrUgkXwKGAcKIiRH4Uoe8LYnAxGkbc0gUSGzZ0Lc9HDox3DMAAFpCeAsneV7wGwYRWLszGYQScQvmoGFBgSJGjqw46QiRHFjmwalgJRlokeum5YTuurqdg%2BkbgkRmgru8AkSxwaYdYlHYggCIMAJ1B-ngT5fAc1knGcSAXHA1wfCW9mEZ8ZlgOimKgsoOn2dRRC0fRrDUJQUCSUGLa0Ox0UAIL5m5%2BApdgpjeeQdmOX8wK4aaqJwLh9h2A4SmaROJHVAqIYuSVHn6vJXqRfR9Jfiy2Venyg7ACRfXuSlXlPiRF72cqOisA6STUB6UlWuwW3JGU82lZ5lXVPhClhiSmj2UGI3RWRMAIpNClUUeqXRSQTCmHAr1vdguX5SRRWuQt1BYbh1UAw5b3Jm910RlGBD2fDcbOcV4PTpiFh4XdD0kWNTITYle0ZvygpzZjp1LVVYCrWT61sIdO23WTV4HY6x3Ux5tMXXdoY%2BuGbMwwTYCRuYeWsP9QbJZ9JHfb9MtekD96FTzZVhDjli2uYBDVNDANo0Gxteoj2CRtGqP2SGTkhuobAIFAyQlLTvk7gF5xWKFdzhXtYuxViCW6dgcvodF6WZcrqscTeTsu%2BVlWGwSTl1SwDWQywLVtWAymqV1PWYQ7WTO9QJQGJ8l0EmLRPMhAyvTZTpFx6XrtJ-TIdM5tjq7W91os2Uxfx2XfNV2bQs3WP93liRT0vaTMMEmHdFfTAP1-Qvi8x9Fw4twnBjHMnCmm5aJ%2BC76FvI9bGyn4Xlp4B1kKDfsHunF7Vw3L7Y9i3PyvL-R3kGDcGjmxNWJF9TYCgDaB%2BKktLKxkrsQcU9ooHiPNIU82l6KoPhOgs8ytFJgEAkAkw4xGAeCYC2WyqBopEOkKwUhV5PCUI3iHAh1wpaQPSCyRhFDaKmRIuwtgnCeAQB4cwo%2BKdbY-DTggDOTUs6tRznnOBYBuohigMlXBT9cJDQIjPMAtcSYh0bkOWej99QrU7kFZmPcRYKX7lzGBqkIGKJQZok8eDVHIO9BfbxeiiKPSNM9fBssPrh1nlxFhAsVagKwe4uAGDsACR0HgEJXoEGNF0TDEif4YBEBbDeO86kcl5IKWwIpm9F4kQAEL5KOOlFG1Cal1MKY06Jp92ln0kezc2lsUZ7VNnbTCGi0EeMhMOIgOgsnV30YY%2BulTeRnBmiREZOCxkWI7mtax3dtq93sZzbaZRVnHgSWeFxrVoq5JaeUxp-N2Y%2BOFn4ikP8gnz1YaEmi4SGLELSZabeJFsEnMSckhAvyKQZLknYqpYAaxEARDxDeKASKwvhVE%2B5ekwAABldCIuiti9gG92lxk6QLLphIJ5IytgMqR6j4kYLrBubyEF6GmCYNMssATCYMmJvM4xSym5OK0hshmQYu4sz2UGBxhzBUTnOTnQFWiup%2BPNk8mZnLm7wmCQsgG-9HqRLBaxPKYCwAKrGdgBELKmC2lQvQgg6BUxsBCPAA22q3oQqQUSgh85YAkMaUisA3r4C2oNTQkoLB4QmSaYQsNjrI2ethkbUlAtelX2pTVJQIZaDpRIMoicAAFMuJAoB4DrJi71whyAev9rM7ldcG78tMc3R2rdaYUUeuYqGmy9pitscg-u2hpV7xHpVbAAAyUdtpHEdqzgAQguX4U1pzJxjonVO2BsrmpzocJuXQ%2BhyDumVRSqFAMxYSylga3VCs15K1dTlWJ0VS3YALSUItHIMSsEoIyRAEivRkrJSmqlJsaWYSza4GAubITPtfQVPN5aP1VunuquZ9aKaNqHi29u3JSLTtBV20V2zxXHstP2xxQ625PhXZO6VOHsBborIujBbJx0DqOjK7Rs6Lk7rzE%2BA97SVWetPRYc9t6vSXrAIrNFi9Ab3pBnmp9haYDFtlBWkgRwf0dPZv%2Bo9fTr63wzSB7N4HH5QcU3gY4044OVvZTXWtRj7ImNmk2kuLtW2JWw%2Bu9juGRXjxVIRvtB0WNOnQy5kdzGB5sYgXRhddKzxMdXdRjzkXONLG4-uu5MN%2BPorVVFEiZ62AXrCSvK968DX-LAOZ%2BTL7TPKY-ap79YASWaeTdp1NQH003wpE4Cy5BjhHBrGcbwPkX7%2BTftYj%2BYVv6zLU66srBlLJZDTIgLc6pcNSNtPI2RKJM6MFceZQy2EqFqI6wSCAxA8mpAIIG58ux3YjcCkscbX8Ir6N-q6sTEmQFGvovOM7F3ZQJDcBJBrNV1v1S2xt7O0U5tbgOxvI7elMKw7zfQsIkyht%2BSOKNh7IVP7kD9ohnLGryJvcK-RD7M2ZNgEeMZKwzAUdo7pmtmRcjmq7bANDqyopDuUX0ynS8Ohkj5IQ4svABwiBcNfBWU4ov4R5VkHAHiZDNjCHMEkXYjh0S-UcLxfigkCDCRvGJYIsBtCwEYKyk3EMGUVQQH8BAbhFesGElhRX-wVeQJtNQWQbHIFO699rrr%2B2afwFN-mNEkIXeMDEjeVTE4sKgnMGUXsmF8C5COD1zIavJfvhDCQOEpgdiZKHH4PNWJvL-HyfrhXVA3esHMNgJwiv8xiT99gcwjLGA8X9wkl3LL4Xa598IbAXfZAIjN73jc-fo%2BOsZHH1IWQyjYAMDaBXjBA%2BWVh0k6Q6ua%2BAOkBRnveUK-RgoyQP4FDQTl5t5H4S7AVdZ72E4eEcA0KQL2Kv-YS4txMHpzoLctAI8P9p9NUlxACE8pRKJiAIAkBvJaQ-wAA5TFAwecQCeAoCY4ZPS0HgOABgH8DvbACDRwSyDnanayXUbAS4FXJCckIZAkSwL3TIfALJWHZ%2BSbdVTkSTGJL7HeSJCjJgPiASP7a8UScSWAZWLIdgeiFgztYDS0E6XmSqazfRDg8Q3QeieQxaduWQikDQ6gVg57dguUVQyQ6KXQjZbQgkXQ7GAgXGJQww4wYw9QjWagaw3GI%2BWgikYLYdIbNgwnFQ11CQgBMjVzCwpJMjCuOwvwowgItQwJZtF2CudwyiCLHRIjDlKIhwmIkwsxRLGQ9rS0Y5LRfQ6tew9QRw6KQo9ZPIvTYZGLcZNHSI%2BifwqxbIsASopdCZHQJI2lUZJda3JlS1Ro6KZorZVo9o%2BlDvZlDcJgbogzMDCDEzYtUtSzAUXwpo6Ilo%2BiUDHNYzBTJYstd9StWYy0bYozDzRYvAGDFYlANY4YjY0YrYwzBYvYy42DQ41JYHGok4p43YqrYtcza424kiEY7tWIkiU454v4szCzd444zrbrXrRAfrRbHwgwjIsorI%2BidfHrPrAbEQZaT49GEMU7EgPXK7QHSXIEyAe40E1okk87O8ck2SOEgkJHX-VEko9Ezgy0QI6KNkjRBndw5CHkQ8YgdUXYYcCARXVXL4FoI5SQaQfHCkaQmEVoQeV5HweHHQ5wt2NUwOeKTUo5CXRkAgTUjGMGU6Z%2BOUueQ0iFO8M03qZw1w78NUvLTaaoMoZke0AABg9LtNNK1IJC8PIyijlP1OxFtONPtMDLCPiLLgiLVJtL9KjIDOSJw1lPVM1QREjNkmjMonGLOVSOtI1OTNzNTJ6LWT6MmMGMTJLKNLLIdNqN6IwU6K7GLKzJzN2DzMzR%2BPOJeOWPeIzLdMNK9J8FIAAFYPTVcSgfBJz6yuzyy5idi%2ByoSrjBzXShNmYpySAxyAAOMoXcqcsuHwAANlLIXMbO%2BPmN%2BNfQBPXLlOHO3J9KPJnJPLKAABZzzYBuzMJsTESEBkTBtQyygVtOzvzFzLR6SyTYBrtJd2zEQwKTTLzlTg84BkcBS-8Mykz5zwLNTecTQlBrA6glAdJxYmQEBUBxMlhygZAwBrBJQgA&label=LYewrgzgpg%2BhYCcDmUEE8YDMQOEA)

### CODE REVIEW

File: lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    },
+    aneglesiaVolume: {
+       kind: "dynamic",
+       deps: ["aneglesiaUsed"],
+       render(data) {
+        if(data.aneglesiaUsed){
+          return {
+          kind: "string",
+          variant: "input",
+          label: "Aneglesia volume"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts:78]: Should this be only numbers, and specify units?

File: lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    brainSurgeryPaxinosCoords: {
+      kind: "dynamic",
+      deps: ["stereotaxUsed","brainSurgeryLocation"],
+      render(data) {
+        if(data.brainSurgeryLocation === "Paxinos coordinates" && data.stereotaxUsed){
+          return {
+            kind: "string",
+            variant: "input",
+            label: "Paxinos coordinates for surgery"
+          }
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts:191]: Please discuss with the users a better way to document this.

This should be a proper structured input, probably multiple fields

Feel free to expand the description here to ensure they get it right.

File: lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        return null
+      }
+    },
+
+    brainSurgeryLocation: createSurgeryDependentField({
+      kind: "string",
+      variant: "select",
+      label: "Brain surgery location",
+      options: {
+        "Bregma": "Bregma",
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts:177]: Is this a thing?

**Review by gdevenyi — CHANGES_REQUESTED** [review]
Should volume be limited to a number, and units be specified?

File: lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts (hunk 1 above)
Comment (david-roper) [reply lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts:78]: should be number of ml, changing now

File: lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts (hunk 2 above)
Comment (david-roper) [reply lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts]: I can base this questions of this paxinos coordinate calculator [here](https://labs.gaidi.ca/mouse-brain-atlas/) and I can set the limit based in what i see in the diagram.

File: lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+    brainSurgeryPaxinosDVCoords: {
+      kind: "dynamic",
+      deps: ["stereotaxUsed","surgeryType"],
+      render(data) {
+        if( data.stereotaxUsed && (data.surgeryType !== 'Ovariectomy' && data.surgeryType !== undefined)){
+          return {
+            kind: "number",
+            variant: "input",
+            label: "DV Paxinos coordinates"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts:209]: coordinate, singular

File: lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    },
+    brainSurgeryPaxinosAPCoords: {
+      kind: "dynamic",
+      deps: ["stereotaxUsed","surgeryType"],
+      render(data) {
+        if( data.stereotaxUsed && (data.surgeryType !== 'Ovariectomy' && data.surgeryType !== undefined)){
+          return {
+            kind: "number",
+            variant: "input",
+            label: "AP Paxinos coordinates"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts:194]: coordinate singular

File: lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+    brainSurgeryPaxinosMLCoords: {
+      kind: "dynamic",
+      deps: ["stereotaxUsed", "surgeryType"],
+      render(data) {
+        if( data.stereotaxUsed && (data.surgeryType !== 'Ovariectomy' && data.surgeryType !== undefined)){
+          return {
+            kind: "number",
+            variant: "input",
+            label: "ML Paxinos coordinates"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-Surgery-Form/index.ts:180]: coordinate singular

**Review by gdevenyi — CHANGES_REQUESTED** [review]
Link wasn't updated.

**Review by gdevenyi — APPROVED** [review]

**Pull request merged**